### ISSUE TITLE
Database migrations

### ISSUE THREAD

**Issue by nauar** [Seekfried/friedybot-d2irc#32]
Implement a system for db migrations to manage releases involving DB changes. A possible option is the usage of [yoyo](https://ollycope.com/software/yoyo/latest/#).

**Comment by nauar** [comment]
Addressing the architectural decision about the DB CI/CD changes raised in issue #26

**Comment by nauar** [comment]
In the last conversation about this topic, we considered these options:

- [yoyo](https://ollycope.com/software/yoyo/latest/)
- [peeweee_migrate](https://github.com/klen/peewee_migrate)
- [peewee-db-evolve](https://github.com/keredson/peewee-db-evolve)

After discussing the advantages and drawbacks of each option, we agreed that for our use case, the 1st two would be a bit overkill. So we will test peewee-db-evolve. We need to test it because [the README states it is only compatible with MySQL and PostgreSQL](https://github.com/keredson/peewee-db-evolve?tab=readme-ov-file#supported-databases).

**Comment by nauar** [comment]
After preparing the tests, we get the following error:

```
Traceback (most recent call last):
  File "/home/arnau/projects/bot/friedybot-d2irc/setupdb_evolve_testv2.py", line 39, in <module>
    main()
  File "/home/arnau/projects/bot/friedybot-d2irc/setupdb_evolve_testv2.py", line 32, in main
    createDatabase()
  File "/home/arnau/projects/bot/friedybot-d2irc/setupdb_evolve_testv2.py", line 8, in createDatabase
    db.evolve([Servers])
  File "/home/arnau/projects/bot/friedybot-d2irc/venv/lib/python3.12/site-packages/peeweedbevolve.py", line 729, in evolve
    to_run = calc_changes(db, ignore_tables=ignore_tables, schema=schema)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arnau/projects/bot/friedybot-d2irc/venv/lib/python3.12/site-packages/peeweedbevolve.py", line 618, in calc_changes
    existing_columns_by_table = get_columns_by_table(db, schema=schema)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arnau/projects/bot/friedybot-d2irc/venv/lib/python3.12/site-packages/peeweedbevolve.py", line 444, in get_columns_by_table
    raise Exception("don't know how to get columns for %s" % db)
Exception: don't know how to get columns for <peewee.SqliteDatabase object at 0x7f6df3fb3fe0>
```

Checking the code of peewee-db-evolve, we see that SQLite databases are not supported:

![image](https://github.com/user-attachments/assets/9630947b-bbd2-4bba-8f47-4b7c2cda4646)

So, we need to check other alternatives:

- [yoyo](https://ollycope.com/software/yoyo/latest/)
- [peewe_migrate](https://github.com/klen/peewee_migrate)
- Manage it ourselves directly in the `setupdb.py`.

**Comment by nauar** [comment]
Solution with [peewe_migrate](https://github.com/klen/peewee_migrate) implemented and it works! Review of PR #36 pending.
